shards/modules/fs/fs.cpp

```diff
@@ -27,6 +27,46 @@ struct FileNotFoundException : public std::runtime_error {
   FileNotFoundException(const std::string &err) : std::runtime_error(err) {}
 };
 
+// Path security helpers
+static bool hasPathTraversal(const fs::path &p) {
```

Comment:
Why do we need to check this specifically, Shouldn't having a base path filter be enough for all sandboxing needs?